### PR TITLE
[WIP] make release-notes

### PR DESCRIPTION
**Pull request by edsantiago**
Use k8s release-notes tool to compile a list of release notes.

This is not what I had hoped for, but I can't take it any further. In particular:
* It brings in a LOT of vendor code that is not actually needed for podman.
* It has to be run manually, by someone with a `GITHUB_TOKEN`. I had hoped this could be run by cron somehow, but that requires magic knowledge of github actions and/or Cirrus that I don't possess.
* It is specific to this branch. It will need to be cherrypicked into v4.1 to use it there. Then, as bugs are fixed or new tool versions brought in, those too will need to be backported. That is horrible. The correct way to do it is to have this release-notes code be maintained as a separate tool  somewhere, and be given just the podman/buildah/skopeo branch tag. I don't know how to do that.
* Results aren't tracked over time. It might be nice to run the tool daily and git-commit the results somewhere; this could make it easier for a casual maintainer to update release notes text and tags (`kind/foo`) gradually over time instead of in a panic rush at release time.

Discussion / ideas welcome.

Submitted as two commits, for ease of review (the first is the real code; second is just vendoring)

```release-note
None
```

### CODE REVIEW

**Comment by edsantiago** [comment]
[Sample run from this morning](https://github.com/edsantiago/libpod/blob/podman-release-notes/release-notes-main.md).

This exposes a number of flaws with this approach:
* multi-line release-note blocks in PRs do not work at all ("podman machine inspect")
* neither do lists (the "pretty table" one; notice how it does not show the PR/contributor link in the first line)
* we need to do a better job of assigning `kind/FOO` labels :grin:

Even with all that, I think this approach continues to show promise.

**Comment by edsantiago** [comment]
Yeah, CI is failing early. I'm not going to resubmit, because my goal is to get eyeballs on this and thence a better all-around solution. (Secondary goal is to have something that, in case of some urgent release this weekend, @mheon could pull & run to get a starting point on release notes).

File: go.mod [hunk 1]
```diff
@@ -72,4 +72,5 @@ require (
 	google.golang.org/protobuf v1.28.0
 	gopkg.in/inf.v0 v0.9.1
 	gopkg.in/yaml.v2 v2.4.0
+	k8s.io/release v0.13.0
```
Comment (Luap99) [inline go.mod:75]: I know this is not for testing but could we vendor this under `test/tools/`, adding this to the main module will force this into all podman module users (pkg/bindings for example) or disto packagers like debian.

**Comment by Luap99** [comment]
Can we just install this tool with `go install` without vendoring it?

**Comment by edsantiago** [comment]
>Can we just install this tool with go install without vendoring it?

That seems totally fine with me, but I don't know Go best practices. What I would prefer is to have all this code in an entirely separate repo, outside of containers/podman-buildah-skopeo, so it doesn't have to be maintained across repos & branches. I don't know what our project guidelines are for that.

**Pull request closed without merging**